src/agent/coverage/src/binary.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use std::collections::{BTreeMap, BTreeSet};
+
+use anyhow::{bail, Result};
+use debuggable_module::{block, path::FilePath, Module, Offset};
+use symbolic::debuginfo::Object;
+use symbolic::symcache::{SymCache, SymCacheConverter};
+
+use crate::allowlist::TargetAllowList;
+
+#[derive(Clone, Debug, Default)]
+pub struct BinaryCoverage {
+    pub modules: BTreeMap<FilePath, ModuleBinaryCoverage>,
+}
+
+#[derive(Clone, Debug, Default)]
+pub struct ModuleBinaryCoverage {
+    pub offsets: BTreeMap<Offset, Count>,
+}
+
+impl ModuleBinaryCoverage {
+    pub fn increment(&mut self, offset: Offset) -> Result<()> {
+        if let Some(count) = self.offsets.get_mut(&offset) {
+            count.increment();
+        } else {
+            bail!("unknown coverage offset: {offset:x}");
+        };
+
+        Ok(())
+    }
+}
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct Count(pub u32);
+
+impl Count {
+    pub fn increment(&mut self) {
+        self.0 = self.0.saturating_add(1);
+    }
+
+    pub fn reached(&self) -> bool {
+        self.0 > 0
+    }
+}
+
+pub fn find_coverage_sites<'data>(
+    module: &dyn Module<'data>,
+    allowlist: &TargetAllowList,
+) -> Result<ModuleBinaryCoverage> {
+    let debuginfo = module.debuginfo()?;
+
+    let mut symcache = vec![];
+    let mut converter = SymCacheConverter::new();
+    let exe = Object::parse(module.executable_data())?;
+    converter.process_object(&exe)?;
+    let di = Object::parse(module.debuginfo_data())?;
+    converter.process_object(&di)?;
+    converter.serialize(&mut std::io::Cursor::new(&mut symcache))?;
+    let symcache = SymCache::parse(&symcache)?;
+
+    let mut offsets = BTreeSet::new();
+
+    // If we wanted to apply an allowlist to function names, this is where we'd do it.
```

Comment:
this comment implies (to me) that the functionality doesn't exist, but it does?

Reply:
Ha, stale comment, good catch!

Reply:
https://github.com/microsoft/onefuzz/pull/2701/commits/cbd6f117c2bb6926f14b97af280ccf44e150a576